### PR TITLE
Move RCCHECK macro definition to types.h

### PR DESCRIPTION
**Pull request by norro**
See https://github.com/ros2/rclc/issues/33

### CODE REVIEW

**Comment by pablogs9** [comment]
I don't recommend that since the Return Code Check varies along with platforms:
- FreeRTOS: https://github.com/micro-ROS/freertos_apps/blob/fcd886ee58a301cafbddbe3efb1443663ae43e26/apps/int32_publisher/app.c#L16
- Linux: https://github.com/micro-ROS/micro-ROS-demos/blob/969d7a109859503ab150525d0c0cb582e6ce0499/rclc/int32_publisher/main.c#L11
- etc

**Comment by norro** [comment]
Hmmm... fair point!

However, the fact that this is different across platforms actually strengthens the motivation for providing a convenience macro for this. Given, one could find a way to abstract from the different platforms.

**Comment by norro** [comment]
what do you think abut shipping alternative implementations for different platforms and then selecting which one to compile depending on the platform? this could be done based on [CMAKE_SYSTEM_NAME](https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_NAME.html) or similar techniques.

**Comment by pablogs9** [comment]
Sometimes CMAKE_SYSTEM_NAME is set to Generic when crosscompiling: [check here](https://github.com/micro-ROS/micro_ros_arduino/blob/65efc0dbf0cef7d5e663a16b71166be3abccb238/extras/library_generation/teensy32_toolchain.cmake#L1). So I strongly discourage having a conditional build based on CMake platform name. Also, I think that `rclc` shouldn't have platform-specific code. For example, in FreeRTOS the function that is run by a thread should not return, because it causes a system crash.

Another point is that I suggest using another name to these macros, because duplicating a define may cause a build error when using `-Wpedatic` or `-Werror` in the compiler flags. And we have a lot of usages of the "userland macros". 

In summary, I'm ok if you agree to add some kind of syntax sugar to ease the users' API, but please do not override these existing ones because review all the occurrences can take a lot of time.

**Comment by JanStaschulat** [comment]
@norro  @pablogs9 Please discuss and decide, where to put the macro or close this pull request by the end of this week.

**Comment by pablogs9** [comment]
My only requirement is not to collide with existent naming. So, @norro what's your point? We can merge or delete if another name is given.

**Comment by norro** [comment]
I understand @pablogs9's concerns. So as this is just a minor convenience macro anyway, I suggest to close this PR.

**Comment by JanStaschulat** [comment]
I leave it up to you guys.
